src/boot/efi/measure.c

```diff
@@ -196,7 +196,8 @@ EFI_STATUS tpm_log_event(uint32_t pcrindex, EFI_PHYSICAL_ADDRESS buffer, size_t
         /* If EFI_SUCCESS is returned, will initialize ret_measured to true if we actually measured
          * something, or false if measurement was turned off. */
 
-        if (pcrindex == UINT32_MAX) { /* PCR disabled? */
+        /* PCR disabled or measurement cannot be done? */
+        if (pcrindex == UINT32_MAX || !tpm_present()) {
```

Comment:
thus just mean we are doing the same checks twice for each call, no? is that really an improvement?